### ISSUE TITLE
981172 is the source of way too many false positives

### ISSUE THREAD

**Issue by Paladin** [SpiderLabs/owasp-modsecurity-crs#607]
After writing exceptions for this rule for too many different cookies found in the wild (Trip Advisor sets at least two cookies that trigger this rule for just one example) I'm prompted to wonder if *anyone* is actually able to use this rule on a production website. These are not cookies my site sets, but rather cookies visitors show up with.

Is there a known list of exceptions to add, or do people really automatically lock out all visitors that happen to also visit Trip Advisor or one of the many other sites that set cookies that trip this? I'm hesitant to turn it off completely, but I've been running this as detect only for weeks and it seems every day I see more cookies that set this off. (I'm up to 15 so far.)

The real problem as far as I'm concerned is once I start writing exceptions, where do I stop? I see so many false positives for this rule anymore my default position on any alert it throws is "there it goes again" and not "caught another one."

**Comment by dune73** [comment]
Hey, don't stomp on one of my favorite rules!

You may want to check out https://www.netnea.com/cms/2015/11/19/dont-let-981172-and-981173-disappear-from-the-core-rules/, though. And then https://www.owasp.org/index.php/OWASP_ModSec_CRS_Paranoia_Mode.

The rule and it's friend 981173 is no longer in the default install for the upcoming CRS3. But it is still there in higher paranoia levels, but with more sane limits, which grow stricter gradually the higher to go with the paranoia level.

TL;DR;: Yes, disable the rule completely if it bugs you. It's a pain in the ass in CRS2.

**Comment by lifeforms** [comment]
We agree that rule 981172 in CRS 2 is a very difficult rule, and its drawbacks do not outweigh the advantages for most users.

In CRS 3 we have introduced the concept of *paranoia levels*. You can pick a paranoia level from 1 to 4. The higher you go, the more rules are activated. Our goal is to remove false positives as much as possible in lower paranoia levels. In CRS 3, we moved this 981172 to paranoia level 3, so only the more advanced users desiring strictness will use it, and most people won't be bothered by it.

CRS 2 is going into legacy mode and we will not do usability fixes to it anymore.  I would definitely recommend testing CRS 3.0.0, we will be releasing RC2 next week, your input would be very welcome!

**Comment by Paladin** [comment]
Not saying the rule doesn't do good. My concern was the Boy Who Cried Wolf syndrome; so many false positives might cause real problems to be let through simply because the threat isn't immediately obvious. (The 15 I mentioned above is counting the ones I was pretty sure were legitimate, and in getting that far I had to work hard at keeping from writing an exception as the default action -- i.e., to continue asking "Why should I let you pass?" instead of "Why shouldn't I let you pass?")

My frustration stems from trying to balance the fact I have no control over what a site manager may write into a cookie for their legitimate use, and how I can block a user simply because they've visited a site that was ineptly (as opposed to maliciously) managed. Drawing lines was never easy for me.

Thanks for the blog post pointer. There's a some food in the related posts area that will take me a while to digest.

About halfway through the process I found myself wishing I could use the result of the rule to redirect  someone to a page that says "Your browser has picked up enough suspicious cookies that you should flush them all before returning to this site, for your own protection as well as everyone else's."

And I concur with closing it. The blog post pointer gives an entry point for a lot of good ideas.

**Comment by dune73** [comment]
I feel your pain. It's shared many people when they think of 981172.

But you touch on a generally more sore spot: unwarranted cookies. Browsers pick up cookies via various ways, but I think public wifi and advertisements play a big role as a source of illegitimate cookies being sent to servers who have never seen the cookie before. And their is no easy ModSec way to clean them out of the request.

There was a user's ML thread in spring or so (Winter? Last Fall?), that tried to come up with a good policy in these situation.
